lambdas/es/indexer/index.py

```diff
@@ -18,6 +18,9 @@
 import nbformat
 from tenacity import retry, retry_if_exception, stop_after_attempt, wait_exponential
 
+from t4_lambda_shared.preview import get_preview_lines, MAX_BYTES, MAX_LINES
+
+
 CONTENT_INDEX_EXTS = [
```

Comment:
i think we want to make this one shared list between preview and parquet? anything we can show in plaintext on a webpage also belongs in elastic... for example after this refactor `.parquet` should just be searchable in elastic. i'm not sure if that means we index the HTML preview or, probably smarter, something more like the pandas `head` of the file. the thing to do might be to parameterize get_preview_lines to return either HTML or text. But as written I think we are still missing the property that "anything we can preview, we can and will index". we're not far though. i think we want to wrap get_preview_lines with something that also handles the binary cases (like Parquet; or also handles IPYNB in the smart way that we do (only code and markdown cells)).

Reply:
The preview endpoint doesn't know file extensions, though - that's done by the catalog. In any case, I'd rather do one thing at a time.